src/Microsoft.SymbolStore/KeyGenerators/KeyGenerator.cs

```diff
@@ -85,11 +86,12 @@ public virtual bool IsDump()
         /// <param name="path">full path of file or binary</param>
         /// <param name="id">id string</param>
         /// <param name="clrSpecialFile">if true, the file is one the clr special files</param>
+        /// <param name="checksum">Checksum of pdb file. May be null.</param>
         /// <returns>key</returns>
-        protected static SymbolStoreKey BuildKey(string path, string id, bool clrSpecialFile = false)
+        protected static SymbolStoreKey BuildKey(string path, string id, bool clrSpecialFile = false, VsPdbChecksum checksum = null)
```

Comment:
I believe all of these should be `IEnumerable<VsPdbChecksum>`

Reply:
These helper functions build just one key at a time so I not sure how they would an enumeration of checksums.

Reply:
A PE file can have more than one checksum, if it does, a result is valid as long as at least one match is correct.

This is for two different scenarios -
1. New hash algorithm: as hash algorithms are deprecated, we may want to list more than one has value to maintain compatibly with old systems while supporting strong cypto.
2. Alternative PDBs: The same binary may have multiple PDB files that are all valid. Today none of these scenarios make sense for portable PDBs, but maybe sometime they will. Scenarios for native Windows PDBs -- you can have (1) a normal PDB (2) a PDB with srcsrv stream injected -and- (3) a "public" PDB (no types or source lines).

Reply:
I think I understand that PE's can have more than one checksum, but the enumeration/validation would have to be done at the higher level in the PEFileKeyGenerator.GetKeys implementation like Jason did unless I misunderstood your point.

Reply:
No, what Jason did in `PEFileKeyGenerator` was incorrect. What it should have done was gather up all the checksums and assigned them to all the keys.